### PR TITLE
optimize: seata-server is developed in idea and console support output logs

### LINKED ISSUE #6284: Seata-server is developed in idea and cannot output logs
- [ ] I have searched the [issues](https://github.com/seata/seata/issues) of this repository and believe that this is not a duplicate.

### Ⅰ. Issue Description
seata-server can't output logs from development console in idea

image
为了在容器中不在控制台输出过多无法滚动的日志，所以在pr https://github.com/apache/incubator-seata/pull/6154 中被修改为当应用启动完成后不再输出到控制台。导致本地开发时控制台日志为空

In order to not output too much unscrollable logging on the console in containers, it was changed in pr https://github.com/apache/incubator-seata/pull/6154 to no longer output to the console when the application has finished starting. This causes the console log to be empty when developing locally

### Ⅱ. Describe what happened

  If there is an exception, please attach the exception trace:

```
Just paste your stack trace here!
```

### Ⅳ. How to reproduce it (as minimally and precisely as possible)

1. xxx
2. xxx
3. xxx

Minimal yet complete reproducer code (or URL to code):

### Ⅵ. Environment:

- JDK version(e.g. `java -version`):
- Seata client/server version:  
- Database version:
- OS(e.g. `uname -a`):
- Others:

### PR DESCRIPTION
**Pull request by PeppaO**
- [ ] I have registered the PR [changes](../changes).

### Ⅱ. Does this pull request fix one issue?

fixes #6284

### CODE REVIEW

File: distribution/bin/seata-server.sh [hunk 1]
```diff
@@ -34,7 +34,7 @@ BASEDIR=`cd "$PRGDIR/.." >/dev/null; pwd`
 BASEDIR=${BASEDIR//"//"/"/"}
 
 . ${BASEDIR}/bin/seata-setup.sh
-JAVA_OPT="${JAVA_OPT} -Dspring.config.additional-location=${BASEDIR}/conf/ -Dspring.config.location=${BASEDIR}/conf/application.yml -Dlogging.config=${BASEDIR}/conf/logback-spring.xml"
+JAVA_OPT="${JAVA_OPT} -Dspring.config.additional-location=${BASEDIR}/conf/ -Dspring.config.location=${BASEDIR}/conf/application.yml -Dlogging.config=${BASEDIR}/conf/logback-spring.xml -Dconsole.bootstrap.output=true"
```
Comment (funky-eyes) [inline distribution/bin/seata-server.sh]:
```suggestion
JAVA_OPT="${JAVA_OPT} -Dspring.config.additional-location=${BASEDIR}/conf/ -Dspring.config.location=${BASEDIR}/conf/application.yml -Dlogging.config=${BASEDIR}/conf/logback-spring.xml -Dproduction.deploy.out=true"
```

File: server/src/main/java/io/seata/server/spring/listener/ServerApplicationListener.java [hunk 2]
```diff
@@ -50,7 +50,7 @@ public boolean supportsEventType(ResolvableType eventType) {
 
     @Override
     public void onApplicationEvent(ApplicationEvent event) {
-        if (event instanceof ApplicationReadyEvent) {
+        if (event instanceof ApplicationReadyEvent && Boolean.parseBoolean(System.getProperty("console.bootstrap.output"))) {
```
Comment (funky-eyes) [inline server/src/main/java/io/seata/server/spring/listener/ServerApplicationListener.java]:
```suggestion
        if (event instanceof ApplicationReadyEvent && Boolean.parseBoolean(System.getProperty("production.deploy.output"))) {
```

File: distribution/bin/seata-server.sh (hunk 1 above)
Comment (PeppaO) [reply distribution/bin/seata-server.sh]: done

File: server/src/main/java/io/seata/server/spring/listener/ServerApplicationListener.java (hunk 2 above)
Comment (PeppaO) [reply server/src/main/java/io/seata/server/spring/listener/ServerApplicationListener.java]: done

File: changes/zh-cn/2.x.md [hunk 3]
```diff
@@ -67,6 +67,7 @@
 - [[#6265](https://github.com/apache/incubator-seata/pull/6265)] 优化在 arm64 上构建前端失败的问题
 - [[#6267](https://github.com/apache/incubator-seata/pull/6267)] 增加 Server 反序列化校验
 - [[#6275](https://github.com/apache/incubator-seata/pull/6275)] 优化.asf.yaml文件中的label格式
+- [[#6291](https://github.com/apache/incubator-seata/pull/6291)] 优化seata-server在开发模式时，控制台未输出完整日志的问题
```
Comment (funky-eyes) [inline changes/zh-cn/2.x.md]:
```suggestion
- [[#6291](https://github.com/apache/incubator-seata/pull/6291)] 优化seata-server在idea等开发工具运行时，控制台未输出完整日志的问题
```

**Review by funky-eyes — APPROVED** [review]
LGTM

**Pull request merged**